internal/cluster/cluster.go

```diff
@@ -178,6 +180,37 @@ func (m *Manager) GetMasterNodesIds(ctx context.Context, c *common.Cluster, db *
 	return m.installationAPI.GetMasterNodesIds(ctx, c, db)
 }
 
+func (m *Manager) tryAssignMachineCidr(cluster *common.Cluster) error {
+	var err error
+	networks := network.GetClusterNetworks(cluster, m.log)
+	if len(networks) == 1 {
+		err = m.db.Model(&common.Cluster{}).Where("id = ?", cluster.ID.String()).Update(&common.Cluster{
+			Cluster: models.Cluster{
+				MachineNetworkCidr: networks[0],
+			},
+			MachineNetworkCidrUpdatedAt: time.Now(),
+		}).Error
+		if err != nil {
+			m.log.WithError(err).Warnf("Set machine cidr for cluster %s", cluster.ID.String())
+		}
+	}
+	return nil
+}
+
+func (m *Manager) autoAssignMachineNetworkCidrs() error {
+	var clusters []*common.Cluster
+	err := m.db.Preload("Hosts", "status not in (?)", []string{models.HostStatusDisabled, models.HostStatusDisconnected, models.HostStatusDiscovering}).
+		Find(&clusters, "vip_dhcp_allocation = ? and machine_network_cidr = '' and status in (?)", true, []string{models.ClusterStatusPendingForInput, models.ClusterStatusInsufficient}).Error
+	if err != nil {
+		m.log.WithError(err).Warn("Query for clusters for machine network cidr allocation")
+		return err
+	}
+	for _, cluster := range clusters {
```

Comment:
Why do you go over *all* the clusters? I fear it might override data. Even if at the moment it doesn't.
Shouldn't it work only after getting an inventory && cidr hasn't been set?